x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/logged_requests/log_search_request.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+export const logSearchRequest = (searchRequest: estypes.SearchRequest): string => {
+  const { body, index, ...params } = searchRequest;
+  const urlParams = Object.entries(params)
+    .reduce<string[]>((acc, [key, value]) => {
+      if (value != null) {
+        acc.push(`${key}=${value}`);
+      }
+
+      return acc;
+    }, [])
+    .join('&');
+
+  const url = `/${index}/_search${urlParams ? `?${urlParams}` : ''}`;
+
+  if (body) {
```

Comment:
The `body` key is going away in 9.0 (https://github.com/elastic/kibana/pull/204858) so we should try to avoid relying on it.

Reply:
I've seen that PR, my idea was to get this one merged first before FF.
Then help to resolve conflicts in  https://github.com/elastic/kibana/pull/204858, since there are multiple places changed in both PRs

Reply:
Won't we have to completely change the way we stringify the query after #204858 ? It seems like resolving the conflicts in that PR will essentially require re-implementing this PR, no?

Reply:
I think it's only this function that needs to be changed.
Utilities that do ES query call have been refactored to not use `body` parameter.
So, I would need handle this